pkg/node/manager/manager.go

```diff
@@ -132,12 +139,21 @@ func (m *manager) AddNode(nodeName string) error {
 	var newNode node.Node
 	var nodeFound bool
 
-	newNode, nodeFound = m.dataStore[k8sNode.Name]
+	_, nodeFound = m.dataStore[k8sNode.Name]
 	if nodeFound {
 		log.Info("node is already processed, not processing add event again")
 		return nil
 	}
 
+	if err = m.CreateCNINodeIfNotExisting(k8sNode); err != nil {
+		m.Log.Error(err, "Failed to create CNINode for k8sNode", "NodeName", k8sNode.Name)
+		// we shouldn't see AlreadyExists error here
+		// double-check here to avoid retry on this error
+		return client.IgnoreAlreadyExists(err)
```

Comment:
Maybe I'm confused, but shouldn't we continue with the logic below line 156 if the node exists? Perhaps we should move the IgnoreAlreadyExists code inside of `CreateCNINodeIfNotExisting`

Reply:
Yeah, makes sense. this is the redundant error check as mentioned above. I have removed it. The inside already has an existing check.